modules/ismp/testsuite/src/mocks.rs

```diff
@@ -372,15 +379,22 @@ impl IsmpHost for Host {
     }
 
     fn allowed_proxy(&self) -> Option<StateMachine> {
-        Some(StateMachine::Kusama(2000))
+        self.consensus_state(*b"prox").map(|_| StateMachine::Kusama(2000)).ok()
```

Comment:
What's the purpose of this? Unnecessary

Reply:
When a proxy state machine is not configured, the assertion in the following test snippet fails (which is undesirable):
```rust
pub fn check_response_source<H: IsmpHost>(host: &H) -> Result<(), &'static str> {
    ...
    assert!(host.allowed_proxy().is_none());
```